search/intermediate_result.cpp

```diff
@@ -282,19 +286,38 @@ void FillDetails(FeatureType & ft, Result::Details & details)
 
   feature::TypesHolder const typesHolder(ft);
 
-  details.m_isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
-  if (details.m_isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
-    details.m_stars = std::min(details.m_stars, osm::MapObject::kMaxStarsCount);
-  else
-    details.m_stars = 0;
+  uint8_t starsCount = 0;
+  bool const isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
+  if (isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), starsCount))
+    starsCount = std::min(starsCount, osm::MapObject::kMaxStarsCount);
+  std::string stars;
+  for (int i = 0; i < starsCount; ++i)
+    stars.append(osm::MapObject::kStarSymbol);
 
   auto const cuisines = feature::GetLocalizedCuisines(typesHolder);
-  details.m_cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
+  auto const cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
 
   auto const roadShields = ftypes::GetRoadShieldsNames(ft);
-  details.m_roadShields = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
-
-  details.m_fee = feature::GetLocalizedFeeType(typesHolder);
+  auto const roadShield = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
+
+  auto const fee = feature::GetLocalizedFeeType(typesHolder);
+    
+  std::string description;
+  
+  if (!stars.empty())
+    description += osm::MapObject::kFieldsSeparator + stars;
```

Comment:
append has less allocations:
```suggestion
    description.append(osm::MapObject::kFieldsSeparator).append(stars);
```